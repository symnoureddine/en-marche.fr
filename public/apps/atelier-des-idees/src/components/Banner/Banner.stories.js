import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import Banner from '.';

const props = {
    name: 'Répondez à notre consultation sur les retraites !',
    started_at: '2019-01-03T16:01:01.670Z',
    ended_at: '2019-01-03T16:01:01.670Z',
    linkLabel: 'Je participe',
    url: 'http://google.fr',
};

storiesOf('Banner', module)
    .addParameters({ jest: ['Banner'] })
    .add('default', () => <Banner {...props} onClose={action('close')} />)
    .add('with extra info', () => (
        <Banner {...props} response_time="2" onClose={action('close')} />
    ));
