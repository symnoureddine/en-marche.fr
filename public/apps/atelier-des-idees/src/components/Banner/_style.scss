.banner {
  position: relative;
  color: white;
  background-color: $dark-blue;

  .banner__main {
    display: flex;
    align-items: center;
    height: 170px;

    > * {
      display: flex;
      flex-direction: column;
    }

    .banner__container__titles {
      z-index: 0;
      position: relative;
      justify-content: center;
      width: 60%;
      height: 100%;
      background-color: $dark-blue;
      padding-left: 200px;

      > * {
        font-weight: normal;
        z-index: 2;
      }
      &__title {
        font-size: 24px;
        margin-bottom: 7px;
      }
      &__subtitle {
        font-size: 16px;
      }
    }

    .banner__container__titles:before {
      content: ' ';
      background: $dark-blue;
      position: absolute;
      top: 0;
      left: 20px;
      right: -40px;
      bottom: 0;
      z-index: 1;
      transform: skew(-10deg, 0deg);
    }

    .banner__container {
      width: 40%;
      height: 100%;
      background: linear-gradient(287deg, $dark-blue, $light-dark-blue);
      justify-content: center;
      align-items: start;
      padding-left: 100px;

      &__link {
        background-color: transparent;
        border-color: $white;
        color: $white;
        font-size: 14px;
        text-decoration-line: none;
      }

      &__close {
        position: absolute;
        top: 6px;
        right: 4px;
      }
    }
  }

  /* responsive */
  @media #{$tablet} {
    .banner__main {
      .banner__container,
      .banner__container__titles {
        width: 50%;
      }

      .banner__container__titles {
        padding-left: 100px;
      }
    }
  }

  @media #{$mobile} {
    .banner__main {
      flex-direction: column;
      align-items: initial;
      height: initial;
      padding: 20px 0;

      .banner__container__titles {
        padding-left: 0;
        width: 100%;

        &__title {
          margin-bottom: 10px;
          font-size: 22px;
        }

        &__subtitle {
          font-size: 14px;
        }
      }

      .banner__container__titles:before {
        content: none;
      }

      .banner__container {
        width: 100%;
        margin-top: 15px;
        padding-left: 0;
        background: none;
        background-color: $dark-blue;

        &__link {
          align-self: flex-start;
          margin: 0;
        }

        &__close {
          top: 0;
          display: flex;
          padding: 0 4px 0 0;
          align-self: flex-end;
          z-index: 2;
        }
      }
    }
  }
}
