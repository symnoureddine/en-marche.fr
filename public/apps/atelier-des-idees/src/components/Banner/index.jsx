import React from 'react';
import PropTypes from 'prop-types';
import icnClose from '../../img/icn_close-white.svg';

class Banner extends React.PureComponent {
    render() {
        return (
            <div className="banner">
                <div className="banner__main l__wrapper">
                    <div className="banner__container__close">
                        <button
                            className="banner__container__close button--transparent"
                            onClick={() => this.props.onClose()}
                        >
                            <img src={icnClose} />
                        </button>
                    </div>
                    <div className="banner__container__titles">
                        <h3 className="banner__container__titles__title">{this.props.name}</h3>
                        <h4 className="banner__container__titles__subtitle">
                            Du{' '}
                            {new Date(this.props.started_at).toLocaleDateString('fr-fr', {
                                day: 'numeric',
                                month: 'long',
                                year: 'numeric',
                            })}{' '}
                            au{' '}
                            {new Date(this.props.ended_at).toLocaleDateString('fr-fr', {
                                day: 'numeric',
                                month: 'long',
                                year: 'numeric',
                            })}
                        </h4>
                    </div>
                    <div className="banner__container">
                        <a href={this.props.url} className="banner__container__link button--secondary" target="_blank">
                            {`${this.props.linkLabel}${
                                this.props.response_time ? ` (${this.props.response_time} MIN)` : ''
                            }`}
                        </a>
                    </div>
                </div>
            </div>
        );
    }
}

Banner.defaultProps = {
    response_time: undefined,
    linkLabel: 'Je participe',
};

Banner.propTypes = {
    response_time: PropTypes.string,
    url: PropTypes.string.isRequired,
    linkLabel: PropTypes.string,
    started_at: PropTypes.string.isRequired,
    ended_at: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    onClose: PropTypes.func.isRequired,
};

export default Banner;
