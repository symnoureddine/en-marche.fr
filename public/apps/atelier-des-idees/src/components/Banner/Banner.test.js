import React from 'react';
import TestRenderer from 'react-test-renderer';
import Banner from '.';

describe('Banner', () => {
    let props;

    beforeEach(() => {
        props = {
            title: 'Répondez à notre consultation sur les retraites !',
            subtitle: 'Du 5 aout 2018 au 29 septembre 2018',
            linkLabel: 'Je participe (2 min)',
            link: 'http://google.fr',
            onClose: () => {},
        };
    });

    it('renders correctly', () => {
        const tree = TestRenderer.create(<Banner {...props} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
