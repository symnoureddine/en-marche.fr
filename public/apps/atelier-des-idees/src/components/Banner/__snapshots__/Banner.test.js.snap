// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner renders correctly 1`] = `
<div
  className="banner"
>
  <div
    className="banner__container__close"
  >
    <button
      className="banner__container__close button--transparent"
      onClick={[Function]}
    >
      <img
        src="/assets/img/icn_close-white.svg"
      />
    </button>
  </div>
  <div
    className="banner__container__titles"
  >
    <h3
      className="banner__container__titles__title"
    >
      Répondez à notre consultation sur les retraites !
    </h3>
    <h4
      className="banner__container__titles__subtitle"
    >
      Du 5 aout 2018 au 29 septembre 2018
    </h4>
  </div>
  <div
    className="banner__container"
  >
    <a
      className="banner__container__link button--secondary"
      href="http://google.fr"
      target="_blank"
    >
      Je participe (2 min)
    </a>
  </div>
</div>
`;
